### PR TITLE
Resolve custom repository classes

### PR DESCRIPTION
**Pull request by mcfedr**
I've taken #18 and fixed it up and rebased on master

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -12,7 +12,13 @@
...
 		"php": "~7.1",
 		"phpstan/phpstan": "^0.11",
-		"nikic/php-parser": "^4.0"
+		"nikic/php-parser": "^4.0",
+		"doctrine/orm": "^2.5",
+		"doctrine/common": "^2.7",
+		"doctrine/annotations": "^1.5",
+		"doctrine/persistence": "^1.1",
+		"doctrine/dbal": "^2.5",
+		"doctrine/event-manager": "^1.0"
```
Comment (mcfedr) [inline composer.json]: I'm quite unconvinced of the need for this - composer-require-checker demands this, but actually if you dont configure 'mapping' you wont need any of this, and if you do, you should have it installed anyway

**Comment by mcfedr** [comment]
Will make sense to rebase this on top of #46 because the code moves into the abstract class

**Comment by ondrejmirtes** [comment]
Hi, the work done by @lcobucci in #40 seems like a more solid base to work on.

I really don't like how `src/DoctrineClassMetadataProvider.php` constructor looks like - it violates the principle "don't do any work in constructor", for example people without the `pdo_sqlite` extension would be out of luck.

File: composer.json (hunk 1 above)
Comment (ondrejmirtes) [reply composer.json]: I don't understand why any of the changes in composer.json were needed.

**Comment by ondrejmirtes** [comment]
Unfortunately there's some unresolved discussion in #39 and #40 is based on that.

**Comment by mcfedr** [comment]
Makes sense, #48 fixes the issues in #39.

If you merge that ill see about fixing up #40 as well.

File: composer.json (hunk 1 above)
Comment (mcfedr) [reply composer.json]: Its because of the code that loads/creates the entity manager, but if we use the approach in #40 this wont be needed.

**Pull request closed without merging**